### PR TITLE
Add a helper test to prepare Manager backup snapshots

### PR DESCRIPTION
**Pull request by mikliapko**
The test prepares backup snapshot for its future use in restore cases (restore benchmark or nemesis). The test populates the cluster with data, runs backup and logs created snapshot details into console. Corresponding jenkinsfile and testcase yamls are added.

Together with test, the new helper method was introduced which extracts compaction strategy string from c-s cmd.

To handle the backup size, the new environmental variable was added (SCT_MGMT_PREPARE_SNAPSHOT_SIZE). It should be specified in Jenkins job.

### Testing

- [x] 🟢 2TB snapshot [job](https://argus.scylladb.com/tests/scylla-cluster-tests/0df88308-4d00-4370-abe1-016484f74b41)

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### CODE REVIEW

**Review by fruch — APPROVED** [review]
LGTM

**Review by vponomaryov — APPROVED** [review]
LGTM

File: mgmt_cli_test.py [hunk 1]
```diff
@@ -258,19 +258,45 @@ def run_verification_read_stress(self, ks_names=None):
...
+        """
+        match = re.search(r"compaction\(strategy=([^)]+)\)", cs_cmd)
+        if match:
+            strategy = match.group(1)
+            if remove_postfix:
+                strategy = re.sub(r"CompactionStrategy$", "", strategy)
+            if lower:
+                strategy = strategy.lower()
+        else:
+            raise ValueError("Compaction strategy not found in cs_cmd.")
```
Comment (soyacz) [inline mgmt_cli_test.py:297]: if strategy is not provided in c-s then default one is used (size tiered for oss and incremental for enterprise)

File: sdcm/sct_config.py [hunk 2]
```diff
@@ -1158,6 +1158,10 @@ class SCTConfiguration(dict):
              env="SCT_MGMT_NODETOOL_REFRESH_FLAGS", type=str,
              help="Nodetool refresh extra options like --load-and-stream or --primary-replica-only"),
 
+        dict(name="mgmt_prepare_snapshot_size",
+             env="SCT_MGMT_PREPARE_SNAPSHOT_SIZE", type=int,
+             help="Size of backup snapshot in Gb to be prepared to be prepared for backup"),
```
Comment (soyacz) [inline sdcm/sct_config.py:1163]: double 'prepared'

File: mgmt_cli_test.py [hunk 3]
```diff
@@ -1252,6 +1278,56 @@ def test_suspend_and_resume(self):
...
+        mgr_cluster = self.ensure_and_get_cluster(manager_tool)
+
+        self.log.info("Run backup and wait for it to finish")
+        backup_task = mgr_cluster.create_backup_task(location_list=self.locations, rate_limit_list=["0"])
+        backup_task_status = backup_task.wait_and_get_final_status(timeout=200000)
+        assert backup_task_status == TaskStatus.DONE, \
+            f"Backup task ended in {backup_task_status} instead of {TaskStatus.DONE}"
+
+        self.log.info("Log snapshot details")
+        self.log.info(
```
Comment (soyacz) [inline mgmt_cli_test.py:1323]: also if this is important, all these things could be sent to Argus as a result (text column) so these details would be visible there without peeking into logs.

**Review by soyacz — COMMENTED** [review]
just few coments - post merge

**Comment by mikliapko** [comment]
> just few coments - post merge

yeah, it makes sense to address these things. I will prepare fixes.

File: mgmt_cli_test.py (hunk 3 above)
Comment (mikliapko) [reply mgmt_cli_test.py:1323]: I managed to post snapshot details to Argus Results, thanks for idea!
image
@soyacz Is there any way for every consecutive run to extend the existing table with a new row, thus, keeping all the snapshots into one table without a need to browse every run in a search of snapshots?

Comment (mikliapko) [reply mgmt_cli_test.py:1323]: Or the data on Results tab is run scoped only?

Comment (soyacz) [reply mgmt_cli_test.py:1323]: yes, it's run scoped only

Comment (mikliapko) [reply mgmt_cli_test.py:1323]: got it, thanks!

**Pull request merged**